lib/miq_environment.rb

```diff
@@ -39,7 +39,7 @@ def self.manageiq_gid
   end
 
   class Command
-    EVM_KNOWN_COMMANDS = %w[apachectl nohup service systemctl].freeze
+    EVM_KNOWN_COMMANDS = %w[apachectl docker nohup podman service systemctl].freeze
```

Comment:
docker has to be done a little differently - perhaps @bdunne can give details, but the tl;dr is that podman _also_ installs a docker shim, so a simple `which` returns true for both.

Reply:
The problem there is that podman-docker and "real" docker are actually different (they don't accept the same set of parameters).  podman-docker is just a fake layer that renames the docker part of the command to podman.  So I expect the question `supports_command?("docker")` should be asking for "real" docker.

Reply:
`podman` has a couple of things that `docker` does not.
In building containers we have to add `--format v2s2` for podman.
When running containers with you may need to add `-v /src:/dest:z` for SELinux labeling.
Some of these things cause errors when passed to `docker`.

Reply:
Right I think that logic would have to live at a higher level, strictly at the command level the docker command is supported.  If it is implemented badly that sounds like an issue with podman (which of course we'd have to work around).

If we had something like `MiqEnvironment.has_docker?` where we can use multiple command checks (similar to is_podified) we could encapsulate that logic.

```
MiqEnvironment::Command.supports_command?("podman") => true
MiqEnvironment::Command.supports_command?("docker") => true
MiqEnvironment.has_podman? => true
MiqEnvironment.has_docker? => false
```

We could add a general `MiqEnvironment.docker_runtime => [docker, podman, kubectl]` per https://github.com/ManageIQ/manageiq/pull/22603#issuecomment-1624233193

Reply:
> podman has a couple of things that docker does not.

Okay then I think we're good here then, if you just want to check for "basic" docker support then anything you'd run with `docker` would work with both podman shim and docker.  If you are passing podman specific options then you already know you need to be checking for podman specifically right?